homeassistant/components/greeneye_monitor/sensor.py

```diff
@@ -301,8 +300,7 @@ def native_value(self) -> float | None:
         if not self._sensor:
             return None
 
-        assert isinstance(self._sensor.temperature, float)
-        return self._sensor.temperature
+        return typing.cast(Optional[float], self._sensor.temperature)
```

Comment:
Is it optional or just a float? Before we asserted it to float.

Reply:
It's optional, and the assert was firing occasionally as a result.